flink-kubernetes/src/main/java/org/apache/flink/kubernetes/KubernetesResourceManagerDriver.java

```diff
@@ -90,7 +92,7 @@ public class KubernetesResourceManagerDriver
     /** Current max pod index. When creating a new pod, it should increase one. */
     private long currentMaxPodId = 0;
 
-    private Optional<KubernetesWatch> podsWatchOpt;
+    private CompletableFuture<KubernetesWatch> podsWatchOptFuture;
```

Comment:
```suggestion
    private CompletableFuture<KubernetesWatch> podsWatchOptFuture =
            FutureUtils.completedExceptionally(
                    new ResourceManagerException(
                            "KubernetesResourceManagerDriver is not initialized."));
```

Reply:
This helps prevent NPE if `podsWatchOptFuture` is accidentally being used before being initialized.